GUI.cpp

```diff
@@ -544,8 +537,8 @@ class GUIElement {
 
 };
 
-std::vector<ALLEGRO_BITMAP*> GUIElement::tiles;
-std::vector<ALLEGRO_BITMAP*> GUIElement::letterTiles;
+std::vector<ALLEGRO_BITMAP*> GUIElement::tiles{ loadTileset("hack/data/art/border-window.png", al_map_rgb(255, 0, 255)) };
```

Comment:
this will leak the allegro bitmaps on unload. allegro _might_ destroy them when allegro shuts down, but you shouldn't rely on that

either:
* wrap the tileset in an object with an explicit destructor that calls `al_destroy_bitmap` on each bitmap in the tileset, or
* wrap each bitmap in a object with an explicit destructor that calls `al_destroy_biamp` on its contained bitmap, and make the tileset a vector of those objects

whichever you choose, make sure you handle rule of 0/3/5 correctly

Reply:
also this might fail because the static constructor will be called when the dll is loaded, which means it'll be called _before_ `al_init_system` is called so allegro probably doesn't know how to load bitmaps at this point

you probably need to use a lazy initialization strategy here